controllers/serviceclaim_controller.go

```diff
@@ -316,12 +357,20 @@ func (r *ServiceClaimReconciler) pushServiceBinding(
 	}
 
 	errs := []error{}
-	l.Info("pushing to application namespaces", "application namespaces", ce.Spec.ApplicationNamespaces)
 	for _, ns := range ce.Spec.ApplicationNamespaces {
-		l.Info("pushing to application namespace", "application namespace", ns)
-		if err := r.pushServiceBindingToNamespace(ctx, cecli, ns, sc, secret); err != nil {
-			errs = append(errs, err)
-			l.Error(err, "error pushing to application namespaces", "application namespace", ns)
+		if nspace == nil {
+			l.Info("pushing to application namespace", "application namespace", ns)
+			if err := r.pushServiceBindingToNamespace(ctx, cecli, ns, sc, secret); err != nil {
+				errs = append(errs, err)
+				l.Error(err, "error pushing to application namespaces", "application namespace", ns)
+			}
+		} else if *nspace == ns {
+			l.Info("pushing to application namespace", "application namespace", ns)
+			if err := r.pushServiceBindingToNamespace(ctx, cecli, ns, sc, secret); err != nil {
+				errs = append(errs, err)
+				l.Error(err, "error pushing to application namespaces", "application namespace", ns)
+			}
+
```

Comment:
this may be simplyfied
```suggestion
		if nspace == nil || *nspace == ns {
			l.Info("pushing to application namespace", "application namespace", ns)
			if err := r.pushServiceBindingToNamespace(ctx, cecli, ns, sc, secret); err != nil {
				errs = append(errs, err)
				l.Error(err, "error pushing to application namespaces", "application namespace", ns)
			}

```

Reply:
Thanks!